test/network_analysis/conservation_laws.jl

```diff
@@ -408,3 +495,107 @@ let
     # @test sol[X[1]][end] ≈ 8.0
     # @test sol[X[2]][end] ≈ 4.0
 end
+
+# Check conservation law elimination warnings (and the disabling of these) for `NonlinearSystem`s
+# and `NonlinearProblem`s.
+let
+    # Create models.
+    rn = @reaction_network begin
+        (k1,k2), X1 <--> X2
+        (k3,k4), X1 + X2 --> 2X3
```

Comment:
I intentionally sat `X1 + X2 --> X3` to try an make the conservation law slightly messier than usual (i.e. to try and avoid always having too trivial systems in the tests).

Reply:
How is this not erroring? It is using a non-reversible reaction as if it is reversible.

Reply:
Fell free to add back another test with your more complicated example, but let’s keep this one since it explores a bunch of remake cases.